src/test/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfigSerializeTest.java

```diff
@@ -0,0 +1,334 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig;
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.amazon.opendistroforelasticsearch.security.support.WildcardMatcher;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.InjectableValues;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
```

Comment:
Any reason to use `XContentBuilder` instead of `ObjectMapper`?

Reply:
It was cleaner to use XContentBuilder to build a json object compared to ObjectMapper. With object mapper, had to make separate child nodes and add them to main json object nodes whereas xcontentbuilder has a more fluid API. 
Main intent was clarity in creating json object for test and there seems to be many ways to do it.

Reply:
Object built with XContentBuilder needs to be converted to String and parsed using ObjectMapper anyway. By using ObjectMapper we can bypass that step.

Reply:
Created issue #656